tools/c7n_azure/c7n_azure/resources/access_control.py

```diff
@@ -96,3 +141,112 @@ class UserRole(RelatedResourceFilter):
 
     RelatedResource = "c7n_azure.resources.access_control.RoleDefinition"
     RelatedIdsExpression = "properties.roleDefinitionId"
+
+
+@RoleAssignment.filter_registry.register('resource-access')
+class ResourceAccessFilter(RelatedResourceFilter):
+    """Filters role assignments that have access to a certain
+    type of azure resource.
+
+    :Example:
+
+        .. code-block:: yaml
+
+            policies:
+               - name: assignments-by-azure-resource
+                 resource: azure.roleassignment
+                 filters:
+                    - type: resource-access
+                      relatedResource: azure.vm
+    """
+
+    schema = type_schema(
+        'resource-access',
+        relatedResource={'type': 'string'},
+        rinherit=RelatedResourceFilter.schema,
+        required=['relatedResource']
+    )
+
+    def __init__(self, data, manager=None):
+        super(ResourceAccessFilter, self).__init__(data, manager)
+        resource_type = self.data['relatedResource']
+        self.factory = Azure.resources.get(
+            resource_type.rsplit('.', 1)[-1])
+
+    def get_related_ids(self, resources, related):
+        related_ids = []
+        for r in related:
+            for resource in resources:
+                if resource['properties']['scope'] in r:
+                    related_ids.append(r)
+                    continue
+
+        return related_ids
+
+    def get_related(self, resources):
+        ctx = ExecutionContext(local_session(Session), self.data, Config.empty())
+        manager = self.factory(ctx, self.data)
+        related = manager.source.get_resources(None)
+        return {r['id']: r for r in related}
+
+    def process_resource(self, resource, related):
```

Comment:
I plan to refactor this, just putting this here for now as a work around to work on graph auth stuff.